src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php

```diff
@@ -545,29 +547,14 @@ private function compileSwitchDefault(bool $hasVars, bool $matchHost): string
                 }
                 break;
             }
-            if (!\$hasRequiredScheme) {
-                if ('GET' !== \$canonicalMethod) {
-                    break;
-                }
-
-                return \$this->redirect(\$rawPathinfo, \$ret['_route'], key(\$requiredSchemes)) + \$ret;
-            }
-
-            return \$ret;
-
-EOF;
-        } else {
-            $code .= <<<EOF
-
-            if (\$requiredMethods && !isset(\$requiredMethods[\$canonicalMethod]) && !isset(\$requiredMethods[\$requestMethod])) {
-                \$allow += \$requiredMethods;
+            if (\$requiredSchemes && !\$hasRequiredScheme) {
```

Comment:
`if (!\$hasRequiredScheme) {` would be enough here

Reply:
fixed